crates/interpreter/src/instructions/memory.rs

```diff
@@ -35,3 +42,35 @@ pub fn msize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
     gas!(interpreter, gas::BASE);
     push!(interpreter, U256::from(interpreter.memory.effective_len()));
 }
+// From EIP-5656 MCOPY
+pub fn mcopy<SPEC: Spec>(interpreter: &mut Interpreter, host: &mut dyn Host) {
+    check!(interpreter, SPEC::enabled(CANCUN));
+    pop!(interpreter, dest, src, len);
+    if len == U256::ZERO {
+        return;
+    }
+
+    let len = as_usize_or_fail!(interpreter, len, InstructionResult::InvalidOperandOOG);
+    gas_or_fail!(interpreter, gas::verylowcopy_cost(len as u64));
+
+    // memory_resize!(interpreter, dest, len);
+    memory_copy_resize(interpreter, host);
+    let dest = as_usize_or_fail!(interpreter, dest, InstructionResult::InvalidOperandOOG);
+
+
+    let src = as_usize_or_fail!(interpreter, src, InstructionResult::InvalidOperandOOG);
+    // Read data with length len from src
+    let binding = interpreter.memory.copy_to_vec(src, len);
+    let data = binding.as_slice()  ;
+    // Write data to dest
+    interpreter.memory.set_data(src, dest, len, data);
+}
+
+pub fn memory_copy_resize(interpreter: &mut Interpreter, _host: &mut dyn Host) {
```

Comment:
This is faulty, and you already have `dest,src, len` in `fn mcopy` so this fn is not needed